packages/codemods/src/codeMods/__tests__/personaToAvatar/componentMod.test.ts

```diff
@@ -0,0 +1,77 @@
+import { Project, SyntaxKind, InterfaceDeclaration, TypeReferenceNode, PropertyAccessExpression } from 'ts-morph';
+import {
+  replacePersonaImport,
+  replaceIPersonaPropsImport,
+  replacePersonaSizeImport,
+} from '../../mods/PersonaToAvatar/PersonaToAvatar.mod';
+import { findJsxTag } from '../../utilities';
+const personaPath = '/**/__tests__/mock/**/persona/**/*.tsx';
+
+describe('Persona component mod tests', () => {
+  let project: Project;
+
+  beforeEach(() => {
+    project = new Project();
+    project.addSourceFilesAtPaths(`${process.cwd()}${personaPath}`);
+  });
+
+  it('can replace persona with avatar', () => {
+    const file = project.getSourceFileOrThrow('mFunction.tsx');
+    replacePersonaImport(file);
+    file.getImportDeclarations().forEach(imp => {
+      imp.getNamedImports().forEach(name => {
+        expect(name.getText()).not.toEqual('Persona');
+      });
+      expect(findJsxTag(file, 'Persona').length).toBe(0);
+    });
+  });
+
+  it('can replace IPersonaProps with AvatarProps', () => {
+    const file = project.getSourceFileOrThrow('mInterface.tsx');
+    replaceIPersonaPropsImport(file);
+    file.getImportDeclarations().forEach(imp => {
+      imp.getNamedImports().forEach(name => {
+        expect(name.getText()).not.toEqual('IPersonaProps');
+      });
+      file.forEachDescendant(val => {
+        switch (val.getKind()) {
+          case SyntaxKind.InterfaceDeclaration: {
+            const tVal = val as InterfaceDeclaration;
+            const struct = tVal.getStructure();
+            expect(
+              (struct.extends as string[])?.some(str => {
+                return str === 'IPersonaProps';
+              }),
+            ).toBe(false);
+            break;
+          }
+          case SyntaxKind.TypeReference: {
+            expect((val as TypeReferenceNode).getText()).not.toEqual('IPersonaProps');
+          }
+        }
+      });
+    });
+  });
+
+  it('can replace PersonaSize with AvatarSize', () => {
+    const file = project.getSourceFileOrThrow('mWithPersonaSize.tsx');
+    replacePersonaSizeImport(file);
+
+    file.forEachDescendant(val => {
+      // I believe that these are really the only two cases that a reference to PersonaSize
```

Comment:
In theory you could do `PersonaSize[something]` (not sure if that's caught by PropertyAccessExpression) or `Object.keys(PersonaSize)`. Also seems like TS or ts-morph ought to have a built-in utility for finding all references to a type within a file.

Reply:
Yes, you can call getReferences or something like that. But this is just the test, I believe that in the actual prototype mod that we use the rename function to rename all the PersonaSizes within the file without iterating through like that.